### ISSUE TITLE
Syntax highlighting not auto-disabled on huge module

### ISSUE THREAD

**Issue by ribasushi** [CPAN-API/metacpan-web#1665]
According to https://github.com/CPAN-API/metacpan-web/issues/1631#issuecomment-169666595 there is some sort of cutoff system disabling SynHi, making page load times bearable. However this mechanism does not kick in for DBI, making the following URI require about a minute to render: https://metacpan.org/source/TIMB/DBI-1.634/DBI.pm#L621

**Comment by ribasushi** [comment]
This also seems to be taking a *long time* to render: https://metacpan.org/source/RIBASUSHI/DBIx-Class-0.082840/lib/DBIx/Class/ResultSet.pm#L1285-1288
@wchristian - can you confirm?

**Comment by wchristian** [comment]
@ribasushi Yes it does, at least on the first load. Subsequent loads go faster.

**Comment by wchristian** [comment]
This is under profiler, instead of ordinary load, so a lot slower, but it shows the JS is doing an inordinate amount of work.

Possible solutions i see are: Detecting huge files and making syntax highlighting default to off on those; checking to see if the syntax highlighter has been made more advanced with recent releases to asynchronously apply, starting from the viewport.

![screenshot 2016-07-21 11 15 53](https://cloud.githubusercontent.com/assets/175467/17017689/b60dd01e-4f34-11e6-8bce-823fcdc4a28e.png)

**Comment by haarg** [comment]
The syntax highlighting library we are using has seen virtually no development.  I have partial code replacing the syntax highlighter with a different library, but I have no idea if it will impact the speed of these cases significantly.  The other option would be to implement server side highlighting.

**Comment by zakame** [comment]
Anyone working on this? @haarg what was the other library you tried to replace SyntaxHighlighter with?

Apparently the SyntaxHighlighter v4 is now out at https://github.com/syntaxhighlighter/syntaxhighlighter (though not tagged, but released in npm as such https://www.npmjs.com/package/syntaxhighlighter)

**Comment by haarg** [comment]
The other syntax highlighter is [Prism](http://prismjs.com/).  The basics are all working, but there are a number of things we have layered on top of the old syntax highlighter that I haven't re-implemented.
